src/core/queryObserver.ts

```diff
@@ -596,27 +596,25 @@ export class QueryObserver<
       return true
     }
 
-    const { notifyOnChangeProps, notifyOnChangePropsExclusions } = this.options
+    const { notifyOnChangeProps } = this.options
 
-    if (!notifyOnChangeProps && !notifyOnChangePropsExclusions) {
+    if (notifyOnChangeProps === 'all') {
       return true
     }
 
-    if (notifyOnChangeProps === 'tracked' && !this.trackedProps.length) {
+    if (!notifyOnChangeProps && !this.trackedProps.length) {
       return true
     }
 
-    const includedProps =
-      notifyOnChangeProps === 'tracked'
-        ? this.trackedProps
-        : notifyOnChangeProps
+    const includedProps = !notifyOnChangeProps
+      ? this.trackedProps
+      : notifyOnChangeProps
```

Comment:
negated ifs are pretty hard to read imo, so is this the same as:

```
    const includedProps = notifyOnChangeProps ?? this.trackedProps
```

?

Reply:
Totally agree on using `??`